File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse15.c

#include <stdio.h>
#include <stdbool.h>

#define ARRAY_SIZE 10

static const int id_sequence[ARRAY_SIZE] = {10, 20, 30, 40, 50, 60, 70, 80, 90, 100};

int getValueAtIndex(int index, bool* valid) {
    *valid = false;
    if (index < 0 || index >= ARRAY_SIZE) {
        return -1;
    }
    *valid = true;
    return id_sequence[index];
}

int getValueWithPrompt(bool* valid) {
    printf("Enter an index value: ");
    int index;
    if (scanf("%d", &index) != 1) {
        while (getchar() != '\\n');
        *valid = false;
        return -1;
    }
    return getValueAtIndex(index, valid);
}

int main() {
    bool valid;
    
    // Test cases
    printf("Test Case 1 - Index 0: %d (Valid: %d)\\n", getValueAtIndex(0, &valid), valid);
    printf("Test Case 2 - Index 5: %d (Valid: %d)\\n", getValueAtIndex(5, &valid), valid);
    printf("Test Case 3 - Index 9: %d (Valid: %d)\\n", getValueAtIndex(9, &valid), valid);
    printf("Test Case 4 - Index -1 (invalid): %d (Valid: %d)\\n", getValueAtIndex(-1, &valid), valid);
    printf("Test Case 5 - Index 10 (out of bounds): %d (Valid: %d)\\n", getValueAtIndex(10, &valid), valid);
    
    return 0;
}

File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse15.py

def get_value_at_index(index, id_sequence):
    """Safely retrieve value at index with bounds checking"""
    if not isinstance(index, int):
        print("Error: Index must be an integer")
        return None
    
    if index < 0 or index >= len(id_sequence):
        print(f"Error: Index out of bounds. Valid range: 0 to {len(id_sequence) - 1}")
        return None
    
    return id_sequence[index]

def get_value_with_prompt(id_sequence):
    """Prompt user for index and return value"""
    try:
        user_input = input("Enter an index value: ").strip()
        
        if not user_input:
            print("Error: Empty input")
            return None
        
        index = int(user_input)
        return get_value_at_index(index, id_sequence)
    except ValueError:
        print("Error: Invalid input. Please enter a valid integer.")
        return None
    except EOFError:
        print("Error: Input stream closed")
        return None

def main():
    id_sequence = [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000]
    
    print("Test Case 1: Valid index 0")
    result1 = get_value_at_index(0, id_sequence)
    print(f"Result: {result1}\\n")
    
    print("Test Case 2: Valid index 5")
    result2 = get_value_at_index(5, id_sequence)
    print(f"Result: {result2}\\n")
    
    print("Test Case 3: Valid index 9 (last element)")
    result3 = get_value_at_index(9, id_sequence)
    print(f"Result: {result3}\\n")
    
    print("Test Case 4: Invalid negative index -1")
    result4 = get_value_at_index(-1, id_sequence)
    print(f"Result: {result4}\\n")
    
    print("Test Case 5: Invalid index 10 (out of bounds)")
    result5 = get_value_at_index(10, id_sequence)
    print(f"Result: {result5}\\n")

if __name__ == "__main__":
    main()
